Wikipedia/Code/WMFAppViewController+Extensions.swift

```diff
@@ -613,10 +613,16 @@ extension WMFAppViewController {
         else { return }
         let wmfLanguage = WMFLanguage(languageCode: language, languageVariantCode: nil)
         let project = WMFProject.wikipedia(wmfLanguage)
+        var userId: Int?
+
+        if let siteURL = dataStore.languageLinkController.appLanguage?.siteURL,
+           let userID = dataStore.authenticationManager.permanentUser(siteURL: siteURL)?.userID {
+            userId = userID
+        }
 
         Task {
             do {
-                let yirDataController = try WMFYearInReviewDataController()
+                let yirDataController = try WMFYearInReviewDataController(userID: userId)
```

Comment:
Here you'll pass `userID` into the next line as a parameter instead of into the init.